src/compiler/binder.ts

```diff
@@ -207,7 +207,8 @@ module ts {
                 exportKind |= SymbolFlags.ExportNamespace;
             }
 
-            if (getCombinedNodeFlags(node) & NodeFlags.Export || (node.kind !== SyntaxKind.ImportDeclaration && isAmbientContext(container))) {
+            if (getCombinedNodeFlags(node) & NodeFlags.Export ||

```

Comment:
It returns the combined flags from the node and its parent nodes up to the top-level declaration node. Was needed by incremental parsing because we can no longer push flags down from the top-level declaration.